### PR TITLE
Add podman troubleshooting

### LINKED ISSUE #2845: Add environment variable to address issues with OCI format vs Docker format when using Podman
### Issue Type

Missing content for an existing feature

### Links to Affected Docs

Might go here?
https://docs.astronomer.io/astro/cli/configure-cli#run-the-astro-cli-using-podman

Could also go here since it is tied to `astro deploy` or `astro dev start`?
https://docs.astronomer.io/astro/cli/astro-deploy

Relevant to both Software and Astro so needs to be updated in both places accordingly

### The Issue or Context

Some customers are hitting an issue when building using Podman that show an error with the following message

`WARN[0010] SHELL is not supported for OCI image format, [/bin/bash -o pipefail -e -u -x -c] will be ignored. Must use `docker` format `

### Fix or Proposed Change

To fix this issue, set the following environment variable `export BUILDAH_FORMAT=docker`.

### Additional Notes

[Zendesk](https://astronomer.zendesk.com/agent/tickets/28995)

[Slack](https://astronomer.slack.com/archives/CL44PA4DB/p1694639835319539) 

### Required Reviewers

_No response_

### PR DESCRIPTION
**Pull request by lzdanski**
Resolves #2845 

@jwitz - thoughts about the tip placement for CLI docs and troubleshooting in **Configure the CLI**?

@nsAstro - Can you please double check that the dockerfile code example I included is formatted correctly for the resolution part?

### CODE REVIEW

File: astro/cli/configure-cli.md [hunk 1]
```diff
@@ -298,6 +298,19 @@ If you receive an error after running `podman ps`, there is likely a problem wit
 
 </Tabs>
 
+### Troubleshoot your configuration
```
Comment (jwitz) [inline astro/cli/configure-cli.md:301]:
```suggestion
### Troubleshoot your Podman configuration
```

File: astro/cli/configure-cli.md [hunk 2]
```diff
@@ -298,6 +298,19 @@ If you receive an error after running `podman ps`, there is likely a problem wit
...
+
+Sometimes, after you configure Podman, the following error occurs when Astro tries to build container images when running `astro deploy` or `astro dev start`.
```
Comment (jwitz) [inline astro/cli/configure-cli.md:303]:
```suggestion
The following error can sometimes occur when the CLI tries to build your Astro Runtime image using Podman:
```

File: astro/cli/configure-cli.md [hunk 3]
```diff
@@ -298,6 +298,19 @@ If you receive an error after running `podman ps`, there is likely a problem wit
...
+Sometimes, after you configure Podman, the following error occurs when Astro tries to build container images when running `astro deploy` or `astro dev start`. 
+
+```bash
+WARN[0010] SHELL is not supported for OCI image format, [/bin/bash -o pipefail -e -u -x -c] will be ignored. Must use `docker` format 
+```
+You can resolve this issue by exporting the `BUILDAH_FORMAT` [environment variable](astro/environment-variables.md) to Podman:
+
+```dockerfile
+export BUILDAH_FORMAT=docker
+```
```
Comment (jwitz) [inline astro/cli/configure-cli.md:312]:
````suggestion

```bash
WARN[0010] SHELL is not supported for OCI image format, [/bin/bash -o pipefail -e -u -x -c] will be ignored. Must use `docker` format 
```

To resolve this issue, run the following command to set the `BUILDAH_FORMAT` environment variable on your machine:

```dockerfile
export BUILDAH_FORMAT=docker
```
````

Comment (jwitz) [reply astro/cli/configure-cli.md:312]: From the original Slack context, I think this configuration is local only, even when you're building an image to deploy to Astro.

File: astro/cli/astro-dev-start.md [hunk 4]
```diff
@@ -35,6 +35,20 @@ astro dev start
 $ astro dev start --env=/users/username/documents/myfile.env
 ```
 
+:::tip
```
Comment (jwitz) [inline astro/cli/astro-dev-start.md:38]:
```suggestion
:::info
```
We should also change the note level in the other troubleshooting topic in `configure-cli`, not sure why that one's a tip

File: astro/cli/astro-dev-start.md [hunk 5]
```diff
@@ -35,6 +35,20 @@ astro dev start
...
+
+If you see the following error when using `astro deploy` and you configured your CLI to work with [Podman](cli/configure-cli.md), you might need to set an additional environment variable to correctly build containers during your code deploys to Astro.
```
Comment (jwitz) [inline astro/cli/astro-dev-start.md:40]: Same edits here as in the other notes.

File: astro/cli/astro-deploy.md [hunk 6]
```diff
@@ -129,10 +129,23 @@ $ astro deploy ckvvfp9tf509941drl4vela81n
 # The CLI automatically selects this Deployment for your Astro project
 $ astro deploy ckvvfp9tf509941drl4vela81n --save
 ```
-
 </TabItem>
 </Tabs>
 
+:::tip
```
Comment (jwitz) [inline astro/cli/astro-deploy.md:135]:
```suggestion
:::info
```

File: astro/cli/astro-deploy.md [hunk 7]
```diff
@@ -129,10 +129,23 @@ $ astro deploy ckvvfp9tf509941drl4vela81n
...
+
+If you see the following error when using `astro deploy` and you configured your CLI to work with [Podman](cli/configure-cli.md), you might need to set an additional environment variable to correctly build containers during your code deploys to Astro.
```
Comment (jwitz) [inline astro/cli/astro-deploy.md:137]: Same edits here as the other notes

**Review by jwitz — CHANGES_REQUESTED** [review]
I added some comments based on how I understood the linked Slack threads. Love the placement of the note in the Podman docs, but it could be overkill in the CLI reference docs? If I get the error, I'm more likely to search for the error text than search the CLI reference docs for troubleshooting config. It would also help us to have a single source of truth in case this changes in the future. 

If you want to keep the note in the CLI docs, I would replace the full troubleshooting step with a link to the troubleshooting Podman docs, saying something like "The functionality of this command might be affected if you use Podman. See [Troubleshoot topic] for more information." This is just my opinion, don't feel strongly if you want to keep it, but wanted to throw it out there as an idea

**Review by nsAstro — DISMISSED** [review]
The changes to the docs look good. Can we make the edits for Astro CLI docs for Software as well? I'm not too familiar with the doc repo structure so I'm guessing it's [here](https://github.com/astronomer/docs/blob/main/software/deploy-cli.md). This issue was identified and resolved in a Software customer's environment.

**Comment by lzdanski** [comment]
@jwitz did an in-person quick talk about adding a note to the Software deploy code page. I think I added the right info to the tips - I'd like to keep that info in the CLI pages just because I can see a use case of seeing an error > searching the command that caused it. But changed it to `info` from `tip`

File: astro/cli/configure-cli.md [hunk 8]
```diff
@@ -298,6 +298,18 @@ If you receive an error after running `podman ps`, there is likely a problem wit
...
+The following error can sometimes occur when the CLI tries to build your Astro Runtime image using Podman:
+
+```bash
+WARN[0010] SHELL is not supported for OCI image format, [/bin/bash -o pipefail -e -u -x -c] will be ignored. Must use `docker` format 
+```
+To resolve this issue, run the following command to set the `BUILDAH_FORMAT` environment variable on your machine:
+```dockerfile
+export BUILDAH_FORMAT=docker
+```
+
```
Comment (jwitz) [inline astro/cli/configure-cli.md:312]:
```suggestion
### Troubleshoot your Podman configuration

The following error can sometimes occur when the CLI tries to build your Astro Runtime image using Podman:

```bash
WARN[0010] SHELL is not supported for OCI image format, [/bin/bash -o pipefail -e -u -x -c] will be ignored. Must use `docker` format 
```

To resolve this issue, run the following command to set the `BUILDAH_FORMAT` environment variable on your machine:

```dockerfile
export BUILDAH_FORMAT=docker
```

```
Small nit but I like to always newline between markdown style types, helps to improve readability and can sometimes resolve formatting issues

**Review by jwitz — APPROVED** [review]
LGTM! Just two v minor suggestions

**Pull request merged**